ci/check-protobuf-against-stable.sh

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+# Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+set -euo pipefail
+
+readonly SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+readonly SRC_DIR="$(dirname "${SCRIPT_DIR}")"
+
+cd "${SRC_DIR}"
+readonly CURRENT_BRANCH="$(git rev-parse --abbrev-ref HEAD)"
+
+set_latest_stable() {
+  local gitTagSearchSpace
+  if [[ "${CURRENT_BRANCH}" == "main" ]]; then
+    gitTagSearchSpace="--no-merged"
+  else
+    gitTagSearchSpace="--merged"
+  fi
+  LATEST_STABLE="$(git tag ${gitTagSearchSpace} | grep -v "snapshot" | sort -V | tail -1)"
+}
+
+cleanup_tmp_files() {
+  rm -rf "${TMP_STABLE_PROTOS_DIR}"
```

Comment:
I'd expect a global variable to be defined at the top-level and before functions start referencing it. Forward references are hard on the reader.

Reply:
This is also done as part of ed1ba01.